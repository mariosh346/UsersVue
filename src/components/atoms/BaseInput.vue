<template>
  <q-input
    :model-value="modelValue"
    :label="label"
    :rules="rules"
    :type="type"
    filled
    v-bind="$attrs"
    class="max-w-lg"
    @update:model-value="$emit('update:modelValue', $event)"
  />
</template>

<script setup lang="ts">
import type { QInputProps } from 'quasar';

defineProps<Omit<QInputProps, "onUpdate:modelValue">>();

defineEmits<{
  (e: 'update:modelValue', value: QInputProps["modelValue"]): void;
}>();
</script>
