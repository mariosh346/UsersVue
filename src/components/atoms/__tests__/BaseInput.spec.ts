import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import BaseInput from "../BaseInput.vue";

describe('BaseInput', () => {
  const inputProps = {
    modelValue: 'test',
    label: 'Input Label',
    rules: []
  };
  let wrapper: ReturnType<typeof shallowMount>;
  const createWrapper = () => shallowMount(BaseInput, {
    props: inputProps
  });

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('renders with a model value', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
