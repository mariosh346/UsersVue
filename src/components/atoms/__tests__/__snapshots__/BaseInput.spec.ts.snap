// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > matches snapshot 1`] = `"<q-input-stub modelvalue="test" label="Input Label" rules="" noerroricon="false" reactiverules="false" lazyrules="false" stacklabel="false" hidehint="false" filled="true" outlined="false" borderless="false" standout="false" square="false" loading="false" labelslot="false" bottomslots="false" hidebottomspace="false" rounded="false" dense="false" itemaligned="false" counter="false" clearable="false" disable="false" autofocus="false" reversefillmask="false" fillmask="false" unmaskedvalue="false" type="text" autogrow="false"></q-input-stub>"`;
